TM1py/Services/ServerService.py

```diff
@@ -194,27 +212,34 @@ def get_transaction_log_entries(self, reverse: bool = True, user: str = None, cu
         :param since: of type datetime. If it doesn't have tz information, UTC is assumed.
         :param until: of type datetime. If it doesn't have tz information, UTC is assumed.
         :param top: int
+        :param tuple: of type dict. Filtervalue as key and comparison operator as value tuple={'Filtervalue1':'eq','Filtervalue2': 'ge'}
```

Comment:
I would suggest calling this parameter `elements` instead of `tuple`. That would be more in line with the existing arguments.
I think there is no harm in including it in the function signature if we set the default value to `None`.
That should make it easier to find and use for people

I can't really think of a good use case for `gt` or `le` for element names (maybe years?). I imagine `eq` and `ne` would be helpful. 
The user might want to choose whether `any` or `all` is used in the filtering. So perhaps we could introduce another argument like `elements_filter_type` or `elements_any_or_all`

Maybe for both values (`gt`/`eq`/... and `any`/`all`), we could offer an enum but also accept a str kinda like we did for element types.
https://github.com/cubewise-code/tm1py/blob/2d45a621e4de9794c239dc2958a6fcad94ed464a/TM1py/Objects/Element.py#L34

Reply:
Thanks @MariusWirtz, I'll add `elements: dict = None`  and `elements_any_or_all: str = 'any'` as parameter and update the log_filter section.
For the comparison operators I've in mind to filter ranges of i.e. accounts with `ge`/`gt`...
so I would stay with the dict of `'filter value':'operator'`

Reply:
I think we don't need the `Tuple/all` expression. This expression is only true if all elements in the collection meet the condition. In this case we have a list with elements and therefore `all` can't be true:
```
Tuple['e1','e2','e3','e4','e5']
Tuple/all(t: t eq 'e1') # always false
```
In the future we'll have a better way to filter. Hubert announced tm1 will support a odata v4.01 feature to request a member of an ordered collection.
With this feature we could use:
`(Tuple/2 eq 'PLAN') # dimension 2 is actual, plan, fc....`

For now I would go with `any` only and update the function when tm1 supports this feature

Reply:
I know what you're thinking about :) 
Unfortunatly /all works in an other way you would expect.
Let's say we have a Tuple with `[e1,e2]` - now your expectation is to select this Tuple by using this expression:
`Tuple/all(e: e eq 'e1' or e eq 'e2')`
But this expression will only return these Tuples:
`[e1,e1] or [e2,e2]`
because /all means every element in the list have to meet the condition.
With the announced feature for a future tm1 version could you select the Tuple with this expression:
`(Tuple/1 eq 'e1') and (Tuple/2 eq 'e2') `
With this feature we're able to refactor the function in tm1py ;-)

Reply:
Understood. `all` is not helpful in thise case.